packages/rest/src/__tests__/integration/rest.server.integration.ts

```diff
@@ -26,32 +26,71 @@ import {
   Request,
   requestBody,
   RequestContext,
+  RestApplication,
   RestBindings,
   RestComponent,
   RestServer,
   RestServerConfig,
-  RestApplication,
 } from '../..';
 const readFileAsync = util.promisify(fs.readFile);
 
 const FIXTURES = path.resolve(__dirname, '../../../fixtures');
 const ASSETS = path.resolve(FIXTURES, 'assets');
 
 describe('RestServer (integration)', () => {
-  it('exports url property', async () => {
-    const server = await givenAServer();
-    server.handler(dummyRequestHandler);
-    expect(server.url).to.be.undefined();
-    await server.start();
-    expect(server)
-      .to.have.property('url')
-      .which.is.a.String()
-      .match(/http|https\:\/\//);
-    await supertest(server.url)
-      .get('/')
-      .expect(200, 'Hello');
-    await server.stop();
-    expect(server.url).to.be.undefined();
+  describe('url', () => {
+    let server: RestServer;
+
+    afterEach('shuts down server', async () => {
+      if (server) {
+        await server.stop();
+        expect(server.url).to.be.undefined();
+        expect(server.rootUrl).to.be.undefined();
+      }
+    });
+
+    it('exports url property', async () => {
+      server = await givenAServer();
+      server.handler(dummyRequestHandler);
+      expect(server.url).to.be.undefined();
+      await server.start();
+      expect(server)
+        .to.have.property('url')
+        .which.is.a.String()
+        .match(/http|https\:\/\//);
+      await supertest(server.url)
+        .get('/')
+        .expect(200, 'Hello');
+    });
+
+    it('includes basePath in the url property', async () => {
+      server = await givenAServer({rest: {basePath: '/api'}});
+      server.handler(dummyRequestHandler);
+      expect(server.url).to.be.undefined();
+      await server.start();
+      expect(server)
+        .to.have.property('url')
+        .which.is.a.String()
+        .match(/http|https\:\/\//);
+      expect(server.url).to.match(/api$/);
+      await supertest(server.url)
+        .get('/')
+        .expect(200, 'Hello');
+    });
+
+    it('exports rootUrl property', async () => {
+      server = await givenAServer();
```

Comment:
Please add a test to verify that when `basePath` is set, `rootUrl` is not affected. That's the main point of `rootUrl` property, right?

I think you should move that new test and this existing test into a new describe block called `rootUrl`.

```ts
describe('RestServer (integration)', () => {
  describe('url', () => {
    // ...
  });

  describe('rootUrl', () => {
    // ...
  });

  // ...
});
```